scripts/idl/matter_idl_parser.py

```diff
@@ -227,12 +251,39 @@ def attribute_with_access(self, args):
 
         return (args[-1], acl)
 
+    def ESCAPED_STRING(self, s):
+        # handle escapes, skip the start and end quotes
+        return s.value[1:-1].encode('utf-8').decode('unicode-escape')
+
+    @v_args(inline=True)
+    def default_value(self, value):
+        return AttributeTransformDefaultValue(value)
+
+    @v_args(inline=True)
+    def attribute_is_callback(self):
+        return AttributeTransformSetCallback()
+
+    @v_args(inline=True)
+    def attribute_is_persist(self):
+        return AttributeTransformSetPersisted()
+
+    def attribute_traits(self, traits):
+        # traits as is as a list
+        return traits
+
     def attribute(self, args):
         # Input arguments are:
-        #   -  tags (0 or more)
-        #   -  attribute_with_access (i.e. pair of definition and acl arguments)
-        tags = set(args[:-1])
-        (definition, acl) = args[-1]
+        #   - tags (0 or more)
+        #   - attribute_with_access (i.e. pair of definition and acl arguments)
+        #   - attribute traits (last element)
+        if type(args[-1]) is tuple:
```

Comment:
You should not expect tuples. You should expect iterables:
```
 from collections.abc import Iterable
....

 if isinstance(args[-1], Iterable):
     XXXXX
```

The unpacking works on iterables in Python 3:

```
definition, acl = args[-1]
```

Reply:
Is this generic python advice or in particular for this code? I know I am generating a tuple here in particular, hence my check,

Reply:
Both. Even if you are generating a tuple, if you care that it's an iterable at runtime, rather than something else, and any iterable would work, then use iterable. Scales better